src/wp-admin/includes/privacy-tools.php

```diff
@@ -770,7 +770,16 @@ function wp_privacy_process_personal_data_export_page( $response, $exporter_inde
 		$accumulated_data = get_post_meta( $request_id, '_export_data_raw', true );
 
 		if ( $accumulated_data ) {
-			$export_data = $accumulated_data;
+			if ( is_array( $accumulated_data ) ) {
+				$export_data = $accumulated_data;
+			} else {
+				wp_send_json_error(
+					sprintf(
+						'Warning: array_merge(): Expected parameter 1 to be an array, %1$s given.',
+						gettype( $accumulated_data )
+					)
+				);
```

Comment:
Preserves the original warning when/if '_export_data_raw' exists but is not an array type.

< PHP 7.3
Warning: array_merge(): Argument #1 is not an array

PHP 7.3-7.4
Warning: array_merge(): Expected parameter 1 to be an array, string given

PHP 8
Fatal error: Uncaught TypeError: array_merge(): Argument #1 must be of type array, string given

See it in action here https://3v4l.org/3EbKY